src/System.Collections/src/System/Collections/Generic/SortedSet.cs

```diff
@@ -1421,8 +1421,7 @@ public void SymmetricExceptWith(IEnumerable<T> other)
             }
             else
             {
-                //need perf improvement on this
-                T[] elements = (new List<T>(other)).ToArray();
+                T[] elements = EnumerableHelpers.ToArray(other);
```

Comment:
The change being made should be a pure win, and without needing additional changes elsewhere.  But I agree it could potentially be taken further using the other overload mentioned.  I suggest we go ahead with this change as-is, and then folks can experiment with further changes on top of it.

With regards to the removal of the comment, I don't know if this addresses the perf concerns the author originally had in mind, but as-is the comment isn't particularly helpful, so removing it makes sense regardless of this change.